resources/providers/awslib/iam/user.go

```diff
@@ -165,55 +162,118 @@ func (p Provider) getMFADevices(ctx context.Context, user types.User) ([]AuthDev
 	return devices, nil
 }
 
-func (p Provider) getUserKeys(ctx context.Context, apiUser types.User) ([]AccessKey, error) {
+func (p Provider) getUserKeys(username string, report map[string]*CredentialReport) []AccessKey {
+	p.log.Debugf("aggregate access keys data for user: %s", username)
+	entry := report[username]
 
-	var keys []AccessKey
-	input := iamsdk.ListAccessKeysInput{
-		UserName: apiUser.UserName,
+	if entry == nil {
+		p.log.Debugf("no entry for user: %s in credentials report", username)
+		return nil
 	}
-	for {
-		output, err := p.client.ListAccessKeys(ctx, &input)
-		if err != nil {
-			return nil, err
-		}
 
-		for _, apiAccessKey := range output.AccessKeyMetadata {
-			output, err := p.client.GetAccessKeyLastUsed(ctx, &iamsdk.GetAccessKeyLastUsedInput{
-				AccessKeyId: apiAccessKey.AccessKeyId,
-			})
+	return []AccessKey{
+		{
+			Active:       entry.AccessKey1Active,
+			LastAccess:   entry.AccessKey1LastUsed,
+			HasUsed:      entry.AccessKey1LastUsed != "N/A",
+			RotationDate: entry.AccessKey1LastRotated,
+		}, {
+			Active:       entry.AccessKey2Active,
+			LastAccess:   entry.AccessKey2LastUsed,
+			HasUsed:      entry.AccessKey2LastUsed != "N/A",
+			RotationDate: entry.AccessKey2LastRotated,
+		},
+	}
+}
 
-			var lastUsed time.Time
-			if err == nil {
-				if output.AccessKeyLastUsed != nil && output.AccessKeyLastUsed.LastUsedDate != nil {
-					lastUsed = *output.AccessKeyLastUsed.LastUsedDate
+func (p Provider) getCredentialReport(ctx context.Context) (map[string]*CredentialReport, error) {
+	var ae smithy.APIError
```

Comment:
Seems a short name that might be harder to understand later.
Consider moving the declaration to https://github.com/elastic/cloudbeat/pull/598/files#diff-32fa8082bf35af27930c2716c52e4c641ffbdfb6ffddfa220baada07bd959fb8R194 as it closer to the usage, or change it to something more explicit (`apiErr`?)